neo/SmartContract/InteropService.cs

```diff
@@ -64,6 +67,52 @@ public static partial class InteropService
         public static readonly uint System_Storage_Delete = Register("System.Storage.Delete", Storage_Delete, 0_01000000, TriggerType.Application);
         public static readonly uint System_StorageContext_AsReadOnly = Register("System.StorageContext.AsReadOnly", StorageContext_AsReadOnly, 0_00000400, TriggerType.Application);
 
+        private static bool CheckItemForNotification(StackItem state)
+        {
+            int size = 0;
+            List<StackItem> items_checked = new List<StackItem>();
+            Queue<StackItem> items_unchecked = new Queue<StackItem>();
+            while (true)
+            {
+                switch (state)
+                {
+                    case Struct array:
+                        foreach (StackItem item in array)
+                            items_unchecked.Enqueue(item);
+                        break;
+                    case Array array:
+                        if (items_checked.All(p => !ReferenceEquals(p, array)))
+                        {
+                            items_checked.Add(array);
+                            foreach (StackItem item in array)
+                                items_unchecked.Enqueue(item);
+                        }
+                        break;
+                    case Boolean _:
+                    case ByteArray _:
+                    case Integer _:
+                        size += state.GetByteLength();
+                        break;
+                    case InteropInterface _:
+                        return false;
+                    case Map map:
```

Comment:
Maybe we cannot assume that "the `Key` in `Map` would never be collections."

```C#
                            foreach (var pair in map)
                            {
                                size += pair.Key.GetByteLength();
                                items_unchecked.Enqueue(pair.Value);
                            }

```

because in introp service we have `System.Runtime.Deserialize`

```C#
                        case StackItemType.Map:
                            Map map = new Map();
                            for (int i = 0; i < placeholder.ElementCount; i++)
                            {
                                StackItem key = stack_temp.Pop();
                                StackItem value = stack_temp.Pop();
                                map.Add(key, value);
                            }
                            item = map;
```